File: COP3504C HW/Python/Cowlab/cowsay.py
import heifer_generator as moo
import sys

if __name__ == "__main__":
    if sys.argv[1] == '-l':
        # cowsay.py -l list all available cows
        print('Cows available: heifer kitteh')
    elif sys.argv[1] == '-n':
        # cowsay.py -n COW MESSAGE print message and then moo.HeiferGenerator.cows[0 or 1] equals COW
        moo.get_cows()
        if sys.argv[2] == "heifer":
            output = " ".join(sys.argv[3:])
            print(output) # MESSAGE printed
            print(moo.HeiferGenerator.cows[0].get_image())
        elif sys.argv[2] == "kitteh":
            output = " ".join(sys.argv[3:])
            print(output) # MESSAGE printed
            print(moo.HeiferGenerator.cows[1].get_image())
        else:
            print(f'Could not find {sys.argv[2]} cow!')
    else:
        # cowsay.py MESSAGE print message and then moo.HeiferGenerator.cows[0]
        moo.get_cows()
        output = " ".join(sys.argv[1:])
        print(output) # MESSAGE printed
        print(moo.HeiferGenerator.cows[0].get_image())

File: COP3504C HW/C++/Lab7/cowsay.cpp
#include <iostream>
#include <string>
#include "HeiferGenerator.h"

int main(int argc, char* argv[])
{
    if (argc == 1)
    {
        return 1;
    }
    auto cows = HeiferGenerator::getCows();
    //0 == heifer
    //1 == kitteh
    //2 == dragon
    //3 == icedragon
    std::string argv1 = argv[1];
    if (argv1 == "-l")
    {
        //List available cows
        std::cout << "Cows available: heifer kitteh dragon ice-dragon\n" << std::endl;
    }
    else if (argv1 == "-n")
    {
        //Print message using animal following -n
        //Need to check that animal following -n is valid
        std::string argv2 = argv[2];
        if (argv2 == "heifer" || argv2 == "kitteh" || argv2 == "dragon" || argv2 == "ice-dragon")
        {
            std::string message;
            for (int i = 3; i < argc; i++)
            {
                //Concatenate the message following "-n [cow] ____"
                message += argv[i];
                if (i != argc - 1)
                {
                    message += " ";
                }
            }
            std::cout << "\n" << message << std::endl;

            //Print cow based on argv[2]
            if (argv2 == "heifer")
            {
                std::cout << cows[0]->getImage() << std::endl;
            }
            else if (argv2 == "kitteh")
            {
                std::cout << cows[1]->getImage() << std::endl;
            }
            else if (argv2 == "dragon")
            {
                std::cout << cows[2]->getImage() << std::endl;
                std::cout << "This dragon can breathe fire.\n" << std::endl;
            }
            else
            {
                std::cout << cows[3]->getImage() << std::endl;
                std::cout << "This dragon cannot breathe fire.\n" << std::endl;
            }

        }
        else
        {
            std::cout << "Could not find " << argv2 << " cow!\n" << std::endl;
        }
    }
    else
    {
        //Print message using default cow
        std::string message;
        for (int i = 1; i < argc; i++)
        {   //Concatenate the message following .\Lab7.exe
            message += argv[i];
            if (i != argc - 1)
            {
                message += " ";
            }
        }

        //Print message and then cow underneath it
        std::cout << "\n" << message << std::endl;
        std::cout << cows[0]->getImage() << std::endl;
    }

    return 0;

}